GoogleForCreators/web-stories-wp
# Pre-Publish Checklist: Only show cards if user has the required permission to fix the issue

## Context

PPC 2.0 - adding role restrictions

## Summary

Some users are not able to fix issues that are shown in the checklist. If a user does not have the required permission, then the card will be hidden from the user.

## Relevant Technical Choices

## To-do

n/a

## User-facing changes



## Testing Instructions

We need to check the checklist based on users who have the `hasUploadMediaAction` versus users who do not.
The `hasUploadMediaAction` lets users upload media to the editor. If a user doesn't have this permission, we will hide specific cards from them.

Roles with the `hasUploadActionPermission`: `Super Admin`, `Administrator`, `Editor`, `Author`
Roles without the `hasUploadActionPermission` (who can still edit stories): `Contributor`

---

We are going to check the following cards:

|Section|Card|Title|
|--|--|--|
|Priority|Story missing poster image|Add poster image|
|Priority|Video missing poster image|Add poster image to every video|
|Priority|Story poster image too small|Increase poster image size to at least 640 x 853px|
|Priority|Story poster image wrong aspect ratio|Correct poster image aspect ratio to 640 x 853px|
|Design|Video resolution too low|Increase video resolution to at least 480p|
|Design|Image resolution too low|Upload a higher resolution poster image to at least 828 x 1792px|
|Accessibility|Video missing captions|Add video captions|

To check all of these changes, you'll need to test with a user with the correct permission.  I'll outline the steps below:

1. Sign into a user that has the `hasUploadActionPermission`.
2. Create a new story
3. Add 4 pages to the story (now you have 5 pages so all checkpoints of the checklist have been hit)
4. Open the checklist. The `Priority` section should be open and you should see the `Add poster image` card in the Priority panel in the checklist  ✅ 
5. In the document panel - add a poster image that is very small (less than 640 by 853px) you should see the poster image too small and wrong aspect ratio cards in the Priority panel in the checklist  ✅ ✅ 
6. Add a video and remove the poster image that is in the design panel. You should see the `Video missing poster image` card in the checklist ✅ 
8. Add a video that is very small with horrible resolution (less than 480px high and 852px wide). You should see `Increase video resolution to at least 480p` ratio card in the Design panel in the checklist ✅
9. Add an image that is very small (less than 828px by 1792px) and resize to make bigger. You should see the `Upload a higher resolution poster image to at least 828 x 1792px` card in the Design panel in the checklist ✅ 
10. Go to the design panel. The video you added may have captions or may not, but if they do then remove them. You should see the `Add video captions` card in the Accessibility panel in the checklist ✅ 


After this, log in as a user that does not have that permission (a `contributor`). Follow the steps above and notice that none of these cards show up.

### QA






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #8257
